### PR TITLE
fix invalid notebook for basis_grid.ipynb

### PR DESCRIPTION
**Pull request by dpanici**
Reuploaded the notebook from before it became invalid on github, after making the changes to it Daniel made in 829f32b
I may be missing some more changes though, @ddudt if you could check your local version of basis_grid.ipynb and check that the changes I made are consistent with yours, which I assume is the most recent version of it?

- fix broken basis_grid.ipynb
- modify github tests action to ignore changes to the docs folder (none of our code tests are affected by what changes inside the docs folder)
- small input file docs page fix

### CODE REVIEW

**Comment by dpanici** [comment]
Something weird is the notebook flickers on github when I view it now... but will wait and see until the docs build, if it looks fine on the docs then it is probably fine.

**Comment by f0uriest** [comment]
Seems like the formatting is still weird in the description of 1D fourier series

**Comment by dpanici** [comment]
Now failing on some issue related to nbsphinx version we use not being compatible with the jinja version we use... will try relaxing the nbsphinx version requirement

**Review by f0uriest — APPROVED** [review]
Formatting looks fine, and the math looks correct but maybe double check with @ddudt if you're not sure

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 sphinx > 3.0.0
-sphinx-rtd-theme == 0.5.0
-sphinx-argparse == 0.2.5
-nbsphinx == 0.8.5
-sphinx_copybutton == 0.3.1
+sphinx-rtd-theme
+sphinx-argparse 
+nbsphinx >0.8.5
+sphinx_copybutton
+ipython_genutils
```
Comment (f0uriest) [inline docs/requirements.txt:6]: see the notes here: https://github.com/ipython/ipython_genutils
Is this necessary? Or do we just need to adjust the required versions for the other stuff?

Comment (dpanici) [reply docs/requirements.txt]: I am not sure what to change version wise besides jinja to revert back to 2.1.2 or whatever we had before. This works for now but I will see if I can get it to work without explicitly relying on this package

**Pull request merged**